moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -95,6 +107,15 @@ struct Contact
 
   /** \brief The type of the second body involved in the contact */
   BodyType body_type_2;
+
+  /** \brief The time until collision */
+  double cc_time;
+
+  /** \brief The continous collision type */
+  ContinouseCollisionType cc_type;
+
+  Eigen::Vector3d nearest_points[2];
+  Eigen::Vector3d cc_nearest_points[2];
```

Comment:
When are `nearest_points` and `cc_nearest_points` different?

Reply:
No they are different, for the CCD case the nearest_points are the ones calculated directly by bullet when checking the casted object against the non-casted object and the cc_nearest_points are calculated by some kind of projection which I can't really follow. See [here](https://github.com/j-petit/moveit/blob/4797bc41755e4143d77cb289c29a87e42a087476/moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/tesseract/bullet_utils.h#L494)

Reply:
I think then we don't need `nearest_points` and `cc_nearest_points` at all and keep using only `pos` of the contact? See the documentation [here](https://pybullet.org/Bullet/BulletFull/classbtManifoldPoint.html#ac4fb334242b89daa97eb635ca69cb01f)

Reply:
Not sure about that, I think TrajOpt might need both points, esp. in the case of a self collision.

Reply:
@ommmid What kind of information do you need of a contact point for TrajOpt?

Reply:
Sorry, not sure at this point but maybe it is good to have both points available in case they are needed.

Reply:
but we can calculate `m_positionWorldOnA = m_positionWorldOnB + m_normalWorldOnB * m_distance1;` so having two points is redundant?

Reply:
That's true, but you might as well keep it if it's already calculated and you're just going to use it again.